parsons/aws/s3.py

```diff
@@ -159,18 +158,33 @@ def list_keys(
 
         continuation_token = None
 
+        # Confirm that prefix has trailing /
+        if not prefix.endswith("/"):
+            prefix = f"{prefix}/"
+
         while True:
             args = {"Bucket": bucket}
+
             if prefix:
                 args["Prefix"] = prefix
+
             if continuation_token:
                 args["ContinuationToken"] = continuation_token
+
             args.update(kwargs)
 
-            resp = self.client.list_objects_v2(**args)
+            try:
+                resp = self.client.list_objects_v2(**args)
 
-            for key in resp.get("Contents", []):
+            except ClientError as e:
```

Comment:
Generally prefer to catch the smallest error. Is this as granular as it gets or is there a more specific auth error?

Reply:
I haven't been able to throw a more granular exception here... seems as soon as the `resp = self.client.list_objects_v2(**args)` request processes and *any* keys are out of the scope of our access it'll raise this error

My initial plan was to isolate the auth component and build a list of files that we have access to, but in a bucket where we have partial access I think we're going to keep bumping up against the same issue. Does that make sense?